src/e2e-test/java/io/cdap/plugin/gcs/stepsdesign/GCSPlugin.java

```diff
@@ -0,0 +1,521 @@
+import io.cdap.e2e.utils.ConstantsUtil;
+import io.cdap.e2e.utils.JsonUtils;
+import io.cdap.e2e.utils.PluginPropertyUtils;
+import io.cdap.e2e.utils.SeleniumDriver;
+import io.cdap.e2e.utils.SeleniumHelper;
+import io.cdap.e2e.utils.StorageClient;
+import io.cdap.plugin.utils.E2ETestConstants;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import org.apache.commons.lang3.StringUtils;
+import org.junit.Assert;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import stepsdesign.BeforeActions;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+/**
+ * GCS Plugin related step design.
+ */
+public class GCSPlugin implements CdfHelper {
+
+  List<String> propertiesSchemaColumnList = new ArrayList<>();
+  Map<String, String> sourcePropertiesOutputSchema = new HashMap<>();
+  String cmekGCS =  PluginPropertyUtils.pluginProp(E2ETestConstants.CMEK_GCS);
+
+  @Given("Open Datafusion Project to configure pipeline")
+  public void openDatafusionProjectToConfigurePipeline() throws IOException, InterruptedException {
+    openCdf();
+  }
+
+  @When("Source is GCS bucket")
+  public void sourceIsGCSBucket() {
+    selectSourcePlugin("GCSFile");
+  }
+
+  @When("Source is BigQuery")
+  public void sourceIsBigQuery() {
+    selectSourcePlugin("BigQueryTable");
+  }
+
+  @When("Target is GCS bucket")
+  public void targetIsGCSBucket() {
+    CdfStudioActions.clickSink();
+    selectSinkPlugin("GCS");
+  }
+
+  @When("Target is BigQuery")
+  public void targetIsBigQuery() {
+    CdfStudioActions.clickSink();
+    selectSinkPlugin("BigQueryTable");
+  }
+
+  @Then("Open GCS source properties")
+  public void openGCSSourceProperties() {
+    openSourcePluginProperties("GCS");
+  }
+
+  @Then("Open GCS Sink Properties")
+  public void openGCSSinkProperties() {
+    openSinkPluginProperties("GCS");
+  }
+
+  @Then("Connect source as {string} and sink as {string} to establish connection")
+  public void connectSourceAndSinkToEstablishConnection(String source, String sink) {
+    connectSourceAndSink(source, sink);
+  }
+
+  @Then("Enter GCS property projectId and reference name")
+  public void enterGCSPropertyProjectIdAndReferenceName() throws IOException {
+    CdfGcsActions.enterProjectId();
+    CdfGcsActions.enterReferenceName();
+  }
+
+  @Then("Enter GCS property path {string}")
+  public void enterGCSPropertyPath(String path) {
+    CdfGcsActions.getGcsBucket(PluginPropertyUtils.pluginProp(path));
+  }
+
+  @Then("Select GCS property format {string}")
+  public void selectGCSPropertyFormat(String format) throws InterruptedException {
+    CdfGcsActions.selectFormat(format);
+  }
+
+  @Then("Toggle GCS property skip header to true")
```

Comment:
Some properties like 'Skip Header' are not applicable to both source and sink. Please make sure that the method name indicates which plugin it's referring to. In this case it's GCSSource

Reply:
method names updated